api/proto/teleport/legacy/types/types.proto

```diff
@@ -1811,8 +1811,10 @@ message AccessRequestFilter {
 message AccessCapabilities {
   // RequestableRoles is a list of existent roles which the user is allowed to request.
   repeated string RequestableRoles = 1 [(gogoproto.jsontag) = "requestable_roles,omitempty"];
+  // NecessaryRolesForResources is a list of the roles necessary to access a given set of resources.
+  repeated string NecessaryRolesForResources = 2 [(gogoproto.jsontag) = "necessary_roles,omitempty"];
```

Comment:
I might call this `ApplicableRolesForResources` instead since not all may be necessary